Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -147,242 +158,140 @@ def _write(self):
-        value = quantize(pair.value, quantization)
-        if rtl and "L" in pair.bidiTypes:
-            # numbers are always shaped LTR even in RTL scripts
-            rtl = False
         valuerecord = ast.ValueRecord(
-            xPlacement=value if rtl else None,
+            xPlacement=pair.value if rtl else None,
             yPlacement=0 if rtl else None,
-            xAdvance=value,
+            xAdvance=pair.value,
             yAdvance=0 if rtl else None,
         )
         return ast.PairPosStatement(
-            glyphs1=pair.side1,
+            glyphs1=cls._convertToFeaAst(pair.side1, ast_cache),
             valuerecord1=valuerecord,
-            glyphs2=pair.side2,
+            glyphs2=cls._convertToFeaAst(pair.side2, ast_cache),
             valuerecord2=None,
             enumerated=enumerated,
         )
 
-    def _makeKerningLookup(
-        self, name, pairs, exclude=None, rtl=False, ignoreMarks=True
-    ):
-        assert pairs
-        rules = []
-        for pair in pairs:
-            if exclude is not None and exclude(pair):
-                self.log.debug("pair excluded from '%s' lookup: %r", name, pair)
-                continue
-            rules.append(
-                self._makePairPosRule(
-                    pair, rtl=rtl, quantization=self.options.quantization
-                )
-            )
-
-        if rules:
-            lookup = ast.LookupBlock(name)
-            if ignoreMarks and self.options.ignoreMarks:
-                lookup.statements.append(ast.makeLookupFlag("IgnoreMarks"))
-            lookup.statements.extend(rules)
-            return lookup
-
-    def _makeKerningLookups(self):
-        cmap = self.makeUnicodeToGlyphNameMapping()
-        if any(unicodeScriptDirection(uv) == "RTL" for uv in cmap):
-            # If there are any characters from globally RTL scripts in the
-            # cmap, we compile a temporary GSUB table to resolve substitutions
-            # and group glyphs by script horizontal direction and bidirectional
-            # type. We then mark each kerning pair with these properties when
-            # any of the glyphs involved in a pair intersects these groups.
-            gsub = self.compileGSUB()
-            dirGlyphs = classifyGlyphs(unicodeScriptDirection, cmap, gsub)
-            directions = self._intersectPairs("directions", dirGlyphs)
-            shouldSplit = "RTL" in directions
-            if shouldSplit:
-                bidiGlyphs = classifyGlyphs(unicodeBidiType, cmap, gsub)
-                self._intersectPairs("bidiTypes", bidiGlyphs)
+    @staticmethod
+    def _convertToFeaAst(
+        side: str | tuple[str, ...],
+        ast_cache: dict[str | tuple[str, ...], ast.GlyphName | ast.GlyphClass],
+    ) -> ast.GlyphName | ast.GlyphClass:
+        """Cache the conversion of a pair name or literal class to the Fea AST,
+        because we'll see the same literal classes over and over."""
```

Comment:
> because we'll see the same literal classes over and over

... so much so that it'd make sense to give them _names_, which would also help with human debugging